### PR TITLE
re-start restoration workers when restarting the application

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#484 

# Overview

- [ ] I have made sure to restart a worker for each known wallet when restarting the wallet server

# Comments

I have not yet find a nice & easy way to test this :thinking:  ...

### CODE REVIEW

File: lib/bech32/src/Codec/Binary/Bech32/Internal.hs [hunk 1]
```diff
@@ -753,7 +753,7 @@ locateErrors residue len
           let p1 = (l_s1 - l_s0 + 1023) `mod` 1023 in
           if (p1 >= len) then [] else
           let l_e1 = l_s0 + (1023 - 997) * p1 in
-          if (l_e1 `mod` 33 > 0) then [] else [p1]
+          [p1 | l_e1 `mod` 33 <= 0]
```
Comment (KtorZ) [inline lib/bech32/src/Codec/Binary/Bech32/Internal.hs]: a bit unrelated but it seems that `hlint` had an update in our CI :|

**Comment by piotr-iohk** [comment]
Perhaps any test would be handy?

**Comment by piotr-iohk** [comment]
ah, ok - just read the comment in description...

>I have not yet find a nice & easy way to test this..

**Comment by piotr-iohk** [comment]
>I have not yet find a nice & easy way to test this

Maybe just looking at the logs after restarting the server? (similarly to logging tests in #474 )

File: lib/core/test/integration/Test/Integration/Framework/DSL.hs [hunk 2]
```diff
@@ -711,6 +714,22 @@ tearDown ctx = do
          Left e -> error $ "deleteAllWallets: Cannot return wallets: " <> show e
          Right s -> s
 
+-- | Wait a booting wallet server to has started. Wait up to 30s or fail.
```
Comment (piotr-iohk) [inline lib/core/test/integration/Test/Integration/Framework/DSL.hs:717]: _Wait a booting wallet server ~to~ has started_ (or _to ~has~ start~ed~_)

**Review by piotr-iohk — APPROVED** [review]
lgtm :+1: 
(only a small typo in comment)

File: lib/core/test/integration/Test/Integration/Framework/DSL.hs (hunk 2 above)
Comment (KtorZ) [reply lib/core/test/integration/Test/Integration/Framework/DSL.hs:717]: Aaaah. Damn. I clicked merge too fast :man_facepalming: 
I'll fix in another PR.

**Pull request merged**